### PR TITLE
Refine validation for rule CRD

### PR DESCRIPTION
**Pull request by sczyh30**
- Refine validation for fields of rule CRD (some fields can be optional) and polish related YAML/controller
- Remove unused logs in data-source controllers
- Add necessary Makefile and `update README.md`

### CODE REVIEW

File: config/rbac/role.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  creationTimestamp: null
+  name: manager-role
+rules:
+- apiGroups:
```
Comment (louyuting) [inline config/rbac/role.yaml:9]: Maybe miss the hotspotRules?

File: controllers/circuitbreakerrules_controller.go [hunk 2]
```diff
@@ -53,18 +53,16 @@ const (
 func (r *CircuitBreakerRulesReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Logger.WithValues("effectiveNs", r.Namespace, "effectiveCrName", r.EffectiveCrName, "req", req.String())
-	log.Info("receive CircuitBreakerRules")
 
 	if req.Namespace != r.Namespace {
-		log.V(int(logging.WarnLevel)).Info("ignore unmatched namespace.")
+		log.V(int(logging.DebugLevel)).Info("ignore unmatched namespace", "namespace", req.Namespace)
```
Comment (louyuting) [inline controllers/circuitbreakerrules_controller.go]: the "req.Namespace" is already in log fixed prefix.

File: controllers/flowrules_controller.go [hunk 3]
```diff
@@ -58,18 +58,16 @@ const (
 func (r *FlowRulesReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Logger.WithValues("effectiveNs", r.Namespace, "effectiveCrName", r.EffectiveCrName, "req", req.String())
-	log.Info("receive FlowRules")
 
 	if req.Namespace != r.Namespace {
-		log.V(int(logging.WarnLevel)).Info("ignore unmatched namespace.")
+		log.V(int(logging.DebugLevel)).Info("ignore unmatched namespace", "namespace", req.Namespace)
```
Comment (louyuting) [inline controllers/flowrules_controller.go]: Same with upon

File: controllers/hotspotrules_controller.go [hunk 4]
```diff
@@ -53,18 +53,16 @@ type HotspotRulesReconciler struct {
 func (r *HotspotRulesReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Logger.WithValues("effectiveNs", r.Namespace, "effectiveCrName", r.EffectiveCrName, "req", req.String())
-	log.Info("receive HotspotRules")
 
 	if req.Namespace != r.Namespace {
-		log.V(int(logging.WarnLevel)).Info("ignore unmatched namespace.")
+		log.V(int(logging.DebugLevel)).Info("ignore unmatched namespace", "namespace", req.Namespace)
```
Comment (louyuting) [inline controllers/hotspotrules_controller.go]: Same with upon

File: controllers/isolationrules_controller.go [hunk 5]
```diff
@@ -46,18 +46,16 @@ type IsolationRulesReconciler struct {
 func (r *IsolationRulesReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Logger.WithValues("effectiveNs", r.Namespace, "effectiveCrName", r.EffectiveCrName, "req", req.String())
-	log.Info("receive IsolationRules")
 
 	if req.Namespace != r.Namespace {
-		log.V(int(logging.WarnLevel)).Info("ignore unmatched namespace.")
+		log.V(int(logging.DebugLevel)).Info("ignore unmatched namespace", "namespace", req.Namespace)
```
Comment (louyuting) [inline controllers/isolationrules_controller.go]: Same with upon

File: controllers/systemrules_controller.go [hunk 6]
```diff
@@ -47,18 +47,16 @@ type SystemRulesReconciler struct {
 func (r *SystemRulesReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	log := r.Logger.WithValues("effectiveNs", r.Namespace, "effectiveCrName", r.EffectiveCrName, "req", req.String())
-	log.Info("receive SystemRules")
 
 	if req.Namespace != r.Namespace {
-		log.V(int(logging.WarnLevel)).Info("ignore unmatched namespace.")
+		log.V(int(logging.DebugLevel)).Info("ignore unmatched namespace", "namespace", req.Namespace)
```
Comment (louyuting) [inline controllers/systemrules_controller.go]: Same with upon

File: controllers/circuitbreakerrules_controller.go (hunk 2 above)
Comment (sczyh30) [reply controllers/circuitbreakerrules_controller.go]: Updated.

File: config/rbac/role.yaml (hunk 1 above)
Comment (sczyh30) [reply config/rbac/role.yaml:9]: This was generated via Makefile. Seems there is something missing regarding the configuration of hotspot rules?

**Pull request merged**